convert2rhel/checks.py

```diff
@@ -57,6 +62,88 @@ def perform_pre_ponr_checks():
     ensure_compatibility_of_kmods()
 
 
+def check_convert2rhel_latest():
+    """Make sure that we are running the latest version of convert2rhel"""
+
+    # get lastest avalible version
+
+    repo_content = (
+        "[convert2rhel]\n"
+        "name=Convert2RHEL Repository\n"
+        "baseurl=https://ftp.redhat.com/redhat/convert2rhel/$releasever/os/\n"
+        "gpgcheck=0\n"
+        "enabled=1\n"
+    )
+
+    repo_dir = os.path.join(utils.TMP_DIR, "convert2rhel_repo")
+    utils.mkdir_p(repo_dir)
+    # repo_dir = os.path.join("/tmp/","convert2rhel_repo")
+    repo_path = os.path.join(repo_dir, "convert2rhel.repo")
+    store_content_to_file(filename=repo_path, content=repo_content)
```

Comment:
@abadger do you think replacing this with the [tempfile](https://docs.python.org/3/library/tempfile.html) is a more secure way? Since we are using this as a temporary file and not something that will be persistent across the whole conversion.

Reply:
I would continue to create the files in `utils.TMPDIR` although you are right that using `tempfile.mkdtemp()` inside of `utils.TMPDIR` might be a little better than hardcoding `"convert2rhel_rep"`.

First of all, you're right that this data is very temporary in nature.  However, in general, that's not the reason that you should use `/tmp.` or `/var/tmp`.  Those should really only be used if you can't depend on some other directory (the user's home directory or, in our case, '/var/lib/convert2rhel/`) or you have to share the file with some other user.  Anytime you operate in `/tmp` or `/var/tmp` it is less secure because those directories are world writable which exposes you to all sorts of race conditions which can be used to attack the system.  The functions in the tempfile stdlib module should be secure against most of those attacks except for a denial of service attack.  Here's an accessible writeup (it doesn't go into the intricacies of what O_CREAT and O_EXCL actually do and do not protect against but it is accurate and pretty easy to understand):  https://owasp.org/www-community/vulnerabilities/Insecure_Temporary_File

Using `tempfile.mkdtemp()` to create repo_dir could be a little better because it generates a name with some random characters in it.  That way if `/var/lib/convert2rhel/convert2rhel.repo` already existed, we wouldn't collide  with whatever was using it.  Using it would look something like this on Line 78:
``` python
repo_dir = tempfile.mkdtemp(prefix="convert2rhel_repo.", dir=utils.TMP_DIR)`
```
If we did that, I we'd need to make sure that utils.TMP_DIR is created at some point before the `mkdtemp()` call.  I can't remember off the top of my head if we're doing that anywhere or if we need to do it here.

Reply:
I have made sure that TMP_DIR is being created in utils, and so I have made this change as well!